src/libYARP_dev/include/yarp/dev/ITransform.h

```diff
@@ -0,0 +1,180 @@
+    * @return true/false
+    */
+    virtual bool     frameExists (const std::string &frame_id) = 0;
+
+    /**
+    Gets a vector containing all the registered frames. 
+    * @param ids the returned vector containing all frame ids
+    * @return true/false
+    */
+    virtual bool    getAllFrameIds (std::vector< std::string > &ids) = 0;
+
+    /**
+     Get the parent of a frame. 
+    * @param frame_id the name of target reference frame
+    * @param parent_frame_id the name of parent reference frame
+    * @return true/false
+    */
+    virtual bool     getParent (const std::string &frame_id, std::string &parent_frame_id) = 0;
+    
+    /**
+     Get the transform between two frames.
+    * @param target_frame_id the name of target reference frame
+    * @param source_frame_id the name of source reference frame
+    * @param transform the transformation matrix from source_frame_id to target_frame_id
+    * @return true/false
+    */
+    virtual bool     getTransform (const std::string &target_frame_id, const std::string &source_frame_id, yarp::sig::Matrix &transform) = 0;
+
+    /**
+     Register a transform between two frames.
+     * @param target_frame_id the name of target reference frame
+    * @param source_frame_id the name of source reference frame
+    * @param transform the transformation matrix from source_frame_id to target_frame_id    
+    * @return true/false
+    */
+    virtual bool     setTransform (const std::string &target_frame_id, const std::string &source_frame_id, const yarp::sig::Matrix &transform) = 0;
+
+    /**
+     Deletes a transform between two frames.
+     * @param target_frame_id the name of target reference frame
+    * @param source_frame_id the name of source reference frame
+    * @return true/false
+    */
+    virtual bool     deleteTransform (const std::string &target_frame_id, const std::string &source_frame_id) = 0;
+
+    /**
+    Transform a point into the target frame.  
+    * @param target_frame_id the name of target reference frame
+    * @param input_point the input point (x y z)    
+    * @param transformed_point the returned point (x y z)
+    * @return true/false
+    */
+    virtual bool     transformPoint (const std::string &target_frame_id, const yarp::sig::Vector &input_point, yarp::sig::Vector &transformed_point) = 0; 
+
+    /**
+     Transform a Stamped Pose into the target frame.
+    * @param target_frame_id the name of target reference frame
+    * @param input_pose the input quaternion (x y z r p y)
+    * @param transformed_pose the returned (x y z r p y)
+    * @return true/false
+    */
+    virtual bool     transformPose (const std::string &target_frame_id, const yarp::sig::Vector &input_pose, yarp::sig::Vector &transformed_pose) = 0;
+    
+    /**
+     Transform a quaternion into the target frame.  
+    * @param target_frame_id the name of target reference frame
+    * @param input_quaternion the input quaternion (x y z w)
```

Comment:
The quaternion convention used in this methods (imaginary/real) is different from the one used in quaternion handling functions in yarp math (real/imaginary , see https://github.com/robotology/yarp/issues/635 ). This could be confusing for YARP users.